src/tribler/core/components/torrent_checker/torrent_checker/tests/test_torrentchecker.py

```diff
@@ -179,9 +179,11 @@ async def test_task_select_tracker(torrent_checker):
     controlled_session.connect_to_tracker = lambda: succeed(None)
 
     torrent_checker._create_session_for_request = lambda *args, **kwargs: controlled_session
+
     result = await torrent_checker.check_random_tracker()
-    assert not result
+    await controlled_session.cleanup()
 
+    assert not result
     assert len(controlled_session.infohash_list) == 1
```

Comment:
On my machine, I got `TypeError: object of type 'NoneType' has no len()`.
The test works if I move `await controlled_session.cleanup()` to the end of the test